test/ipfs.spec.js

```diff
@@ -107,20 +107,37 @@ describe('IPFS', () => {
       })
       const res = await chai
         .request(server)
-        .post(`/api/pin/${peerId}/${x}/${y}`)
+        .post(`/api/pin/${peerId}/${ipfs}/${x}/${y}`)
       expect(res.status, 'Expect status 200').to.be.equal(200)
       expect(
         JSON.stringify(res.body),
         'Expect body to pinning success'
       ).to.be.equal(resExpected)
     })
 
+    it('should not pin files if ipfs expected is different from resolved', async () => {
+      const expectedIPFS = '1'
+      const resExpected = JSON.stringify({
+        error: `The IPFS resolved doesn't match with ${expectedIPFS}. Please wait a few minutes and pin again.`
+      })
+      const res = await chai
+        .request(server)
+        .post(`/api/pin/${peerId}/${expectedIPFS}/${x}/${y}`)
+      expect(res.status, 'Expect status 404').to.be.equal(404)
+      expect(
+        JSON.stringify(res.body),
+        `Expect body The IPFS resolved doesn\'t match with ${expectedIPFS}. Please wait a few minutes and pin again.`
+      ).to.be.equal(resExpected)
+    })
+
     it('should not pin files because peer has not a valid format', async () => {
       connectPeer.restore()
       const resExpected = JSON.stringify({
         error: 'Invalid peerId: peerId'
       })
-      const res = await chai.request(server).post(`/api/pin/peerId/${x}/${y}`)
+      const res = await chai
+        .request(server)
+        .post(`/api/pin/peerId/${ipfs}/${x}/${y}`)
```

Comment:
should be ${peerId}?

Reply:
Nope cause the test is `should not pin files because peer has not a valid format`

Reply:
👍